### PR TITLE
[FC-41933] update the mailserver docs to show an example nix configuration

### PR DESCRIPTION
**Pull request by PhilTaken**
@flyingcircusio/release-managers

Internal ticket: FC-41933

Update the documentation for the mailserver role as well as fixing capitalization for Rspamd.

## Release process

- [x] Created changelog entry using `./changelog.sh`

## PR release workflow (internal)

- [x] PR has internal ticket
- [x] internal issue ID (PL-…) part of branch name
- [x] internal issue ID mentioned in PR description text
- [ ] ticket is on Platform agile board
- [ ] ticket state set to *Pull request ready*
- [ ] if ticket is more urgent than within the next few days, directly contact a member of the Platform team

## Design notes

- [ ] Provide a feature toggle if the change might need to be adjusted/reverted quickly depending on context. Consider whether the default should be `on` or `off`. Example: rate limiting.
- [x] All customer-facing features and (NixOS) options need to be discoverable from documentation. Add or update relevant documentation such that hosted and guided customers can understand it as well.

## Security implications

- [ ] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
- [ ] Security requirements tested? (EVIDENCE)

### CODE REVIEW

File: doc/src/mailserver.md [hunk 1]
```diff
@@ -97,15 +96,16 @@ There must always be exactly one domain with the primary option set.
...
-file into the appropriate DNS zones and don't forget to check reverses.
+After running the above command, a newly-generated file {file}`/etc/local/mail/dns.zone`
+will contain all necessary DNS records for your mail server.
+Insert the records contained within the file into the appropriate DNS zones and
+don't forget to check PTR records for reverse DNS lookups.
 
 ## How do I create users?
 
-Edit {file}`/etc/local/mail/users.json` to add user accounts. Example:
+Edit the file {file}`/etc/local/mail/users.json` to add user accounts. Example:
```
Comment (zagy) [inline doc/src/mailserver.md]: Hm … the users can be defined via nix, too, can't they? Can we document both?

Comment (PhilTaken) [reply doc/src/mailserver.md]: Both are documented now

File: doc/src/mailserver.md [hunk 2]
```diff
@@ -65,28 +66,26 @@ Incorrect DNS setup is the most frequent source of delivery problems. Let our
 :::
 
 If you choose to use the Roundcube webmail UI by adding the `webmailHost`
-setting, like in the example, make sure to enable a `postgresql` role on the
-machine because Roundcube needs it to store settings. Just use the newest
-version that is available at the moment.
+setting like in the example, make sure to enable a `postgresql` role on the
+machine because Roundcube needs it to store it's settings. We recommend you
```
Comment (osnyx) [inline doc/src/mailserver.md]:
```suggestion
machine because Roundcube needs it to store its settings. We recommend you
```

**Review by osnyx — CHANGES_REQUESTED** [review]

**Review by osnyx — APPROVED** [review]
I'd like to get confirmation on whether we intend to deprecate json config for maps. Still merging this for now, as existing json config is not broken by just changing the docs.

@zagy @PhilTaken

**Comment by osnyx** [comment]
rebased onto fc-24.05-dev, hoping this fixes the gitlab test.

**Pull request merged**